facility-all-programs.ipynb

```diff
@@ -81,32 +99,166 @@
+    "            echo_flag = self.echo_type + '_FLAG'\n",
+    "        my_echo_ids = echo_data[ echo_data[ echo_flag ] == 'Y' ][ echo_id ]\n",
+    "        return my_echo_ids\n",
+    "        \n",
+    "    def has_echo_flag( self, echo_data ):\n",
+    "        if ( self.echo_type == 'SDWA' ):\n",
+    "            echo_flag = 'SDWIS_FLAG'\n",
+    "        else:\n",
+    "            echo_flag = self.echo_type + '_FLAG'\n",
+    "        my_echo_data = echo_data[ echo_data[ echo_flag ] == 'Y' ]\n",
+    "        return len( my_echo_data ) > 0\n",
+    "        \n",
+    "region_field = { \n",
+    "    'County': { \"field\": 'FAC_COUNTY' },\n",
+    "    'State': { \"field\": 'FAC_STATE' },\n",
+    "    'Congressional District': { \"field\": 'FAC_DERIVED_CD113' },\n",
+    "    'Zip Code': { \"field\": 'FAC_DERIVED_ZIP' }\n",
+    "}\n",
     "\n",
-    "zip_widget = widgets.IntText(\n",
-    "    value=1,\n",
-    "    description='Zip Code:',\n",
+    "select_region_widget = widgets.Dropdown(\n",
+    "    options=region_field.keys(),\n",
+    "    value='County',\n",
+    "    description='Region of interest:',\n",
     "    disabled=False\n",
     ")\n",
-    "display( zip_widget )"
+    "display( select_region_widget )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Select the state of interest.  If you are interested in a Zip code analysis, this can be skipped."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "states = [\"AL\", \"AK\", \"AZ\", \"AR\", \"CA\", \"CO\", \"CT\", \"DC\", \"DE\", \"FL\", \"GA\", \n",
+    "          \"HI\", \"ID\", \"IL\", \"IN\", \"IA\", \"KS\", \"KY\", \"LA\", \"ME\", \"MD\", \n",
+    "          \"MA\", \"MI\", \"MN\", \"MS\", \"MO\", \"MT\", \"NE\", \"NV\", \"NH\", \"NJ\", \n",
+    "          \"NM\", \"NY\", \"NC\", \"ND\", \"OH\", \"OK\", \"OR\", \"PA\", \"RI\", \"SC\", \n",
+    "          \"SD\", \"TN\", \"TX\", \"UT\", \"VT\", \"VA\", \"WA\", \"WV\", \"WI\", \"WY\"]\n",
+    "dropdown_state=widgets.Dropdown(\n",
+    "    options=states,\n",
+    "    value='NY',\n",
+    "    description='State:',\n",
+    "    disabled=False,\n",
+    ")\n",
+    "output_state = widgets.Output()\n",
+    "my_state = \"\"\n",
+    "\n",
+    "def dropdown_state_eventhandler( change ):\n",
+    "    output_state.clear_output()\n",
+    "    value = change.new\n",
+    "    with output_state:\n",
+    "        display( change.new )\n",
+    "            \n",
+    "dropdown_state.observe( dropdown_state_eventhandler, names='value')\n",
+    "display( dropdown_state )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### If you are looking at a county or congressional district, after selecting a state from the dropdown above, run this cell to pick your region.  \n",
+    "### Run this cell for county, congressional district or zip code regions.  "
```

Comment:
I think the instructions here can be clearer. What if we got rid of this sentence: "If you are looking at a county or congressional district, after selecting a state from the dropdown above, run this cell to pick your region." The two after it seem to cover it.